EngineLayer/GlycoSearch/Glycan.cs

```diff
@@ -5,9 +5,70 @@
 using System;
 using Proteomics;
 using MassSpectrometry;
-
+using System.Dynamic;
+using System.Reflection.Metadata;
+using System.Text;
 namespace EngineLayer
 {
+    public class Monosaccharide
```

Comment:
I think that you need to check if all of the elements are included (4). That names and Ids have not been reused. That nothing is of the wrong type (double instead of int). What happens if you get to a line that has been incorrectly entered? crash?